iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/yaml/SpringConfigYamlConverter.java

```diff
@@ -0,0 +1,169 @@
+    commentsPerProfile.addAll(YamlTreeMetadata.Builder.comments(node));
+  }
+
+  @Override
+  public Stream<TupleBuilder> convertMapping(MappingNode mappingNode) {
+    return mappingNode.getValue().stream().flatMap(this::convertTuple);
+  }
+
+  @Override
+  public Stream<TupleBuilder> convertScalar(ScalarNode scalarNode) {
+    TextRange range = YamlTreeMetadata.Builder.range(scalarNode);
+    var tuple = new TupleBuilder();
+    tuple
+      .addKeyTextRange(range)
+      .addValue(scalarNode.getValue())
+      .addValueTextRange(range);
+    return Stream.of(tuple);
+  }
+
+  @Override
+  public Stream<TupleBuilder> convertTuple(NodeTuple tuple) {
+    if (!(tuple.getKeyNode() instanceof ScalarNode)) {
+      throw new ParseException("Unexpected non-scalar key node in tuple", null, null);
+    }
+
+    // comments for the valueNode are handled during the convert call
+    addComments(tuple.getKeyNode());
+
+    String keyPrefix = ((ScalarNode) tuple.getKeyNode()).getValue();
+    boolean valueIsScalar = tuple.getValueNode() instanceof ScalarNode;
+
+    return convert(tuple.getValueNode())
+      .map((TupleBuilder childTuple) -> {
+        if (valueIsScalar) {
+          return childTuple.addKeyTextRange(YamlTreeMetadata.Builder.range(tuple.getKeyNode()));
+        }
+        return childTuple;
+      })
+      .map(childTuple -> childTuple.prefixKeyDelimited(keyPrefix, tuple.getValueNode().getClass()));
+  }
+
+  @Override
+  public Stream<TupleBuilder> convertSequence(SequenceNode sequenceNode) {
+    return IntStream
+      .range(0, sequenceNode.getValue().size())
+      .mapToObj(index -> {
+        var node = sequenceNode.getValue().get(index);
+        var keyPrefix = "[%d]".formatted(index);
+        return convert(node)
+          .map((TupleBuilder childTuple) -> childTuple.prefixKeyDelimited(keyPrefix, node.getClass()));
+      })
+      .flatMap(Function.identity());
+  }
+
+  public static class TupleBuilder {
+    private String key = "";
+    private TextRange keyTextRange;
+    private String value;
+    private TextRange valueTextRange;
+
+    public TupleBuilder addKeyTextRange(TextRange keyTextRange) {
+      this.keyTextRange = keyTextRange;
+      return this;
+    }
+
+    public TupleBuilder addValue(String value) {
+      this.value = value;
+      return this;
+    }
+
+    public TupleBuilder addValueTextRange(TextRange valueTextRange) {
+      this.valueTextRange = valueTextRange;
+      return this;
+    }
+
+    public Tuple build() {
+      var keyToken = new SyntaxTokenImpl(key, keyTextRange);
+      var valueToken = new SyntaxTokenImpl(value, valueTextRange);
+      return new TupleImpl(new ScalarImpl(keyToken), new ScalarImpl(valueToken));
+    }
+
+    public TupleBuilder prefixKeyDelimited(String prefix, Class<? extends Node> followingNodeType) {
+      var formatString = "%s.%s";
+      if (followingNodeType == ScalarNode.class || followingNodeType == SequenceNode.class) {
+        formatString = "%s%s";
```

Comment:
I don't quite get why we shoulndn't delimit for Scalars

Reply:
In this case the current key is empty. If we would put a delimiter for it the resulting key value pair would look like:
`foo.bar.=value`